### PR TITLE
Add new category fs for file systems

### PR DESCRIPTION
**Pull request by clausecker**
This patch set adds a new category *fs* for file systems and file system utilities.  The new category is populated with some 126 ports (that's more than *x11-clocks*!) that look like they might be file-system related.

I was motivated to add this new category when I noticed that many FUSE file systems were shipped as ports named sysutils/fusefs-$foo, making *sysutils/fusefs-* a pseudo-category for FUSE filesystems.  If that's the (anti) pattern, why not make it offical and add a true category for file system ports?  Turns out there are a lot more than one might think.

I hope that with this move, we can reduce the load on *devel* and *sysutils* (the two most popular misspellings of *misc*) and make file-system related ports easier to find.

If accepted, this'll be the first new physical category since c5978f0c of 2007-05-19 to be added.

For a full list of affected ports, see the list of commits attached to this PR.  If you would like to not have your port moved to *fs*, please comment or send me an email.

Note that this PR is still wip and will be reviewed, exp-ran, etc. before being committed.

### CODE REVIEW

**Comment by metalefty** [comment]
I personally like that idea. That sounds reasonable because there are more ports than x11-clocks. BTW, I like filesystem(s) more than fs.

**Comment by rosorio** [comment]
Hi',El 30 sept 2024 2:30, metalefty ***@***.***> escribió:
I personally like that idea. That sounds reasonable because there are more ports than x11-clocks. BTW, I like filesystem(s) more than fs.Same opinion  +1Rodrigo

**Comment by Jehops** [comment]
Robert, as we discussed in Dublin, I generally think creating a new `fs` category for filesystems is a good idea.  However, I'm on the fence about moving _utilities_ such as `sysutils/zrepl` or `sysutils/zfs-snapshot-mgmt`.  If others agree that also moving those ports is beneficial, then we should make it clear in the comment in `fs/Makefile` that utilities related to filesystems also belong there.

Here are a few more for you:

1.
```
commit 230cb49ccce5a79985875ad16e36da0b8f9e8c52
Author:     Emanuel Haupt <ehaupt@FreeBSD.org>
AuthorDate: 2024-10-02 07:52:13 +0000
Commit:     Emanuel Haupt <ehaupt@FreeBSD.org>
CommitDate: 2024-10-02 07:52:32 +0000

    sysutils/fusefs-mergerfs: Add new port
    
    mergerfs is a union filesystem geared towards simplifying storage and
    management of files across numerous commodity storage devices. It is
    similar to mhddfs, unionfs, and aufs.
```
2. (If we decide fs utilities should also be moved)
```
% cat sysutils/zap/pkg-descr
Maintain and replicate ZFS snapshots

zap automates the management of zfs snapshots.  With a few, simple
crontab entries, it can be used to create a comprehensive zfs backup
system.  There are no configuration files.  All parameters are supplied
on the command line or in zfs properties and all snapshot information
is stored in snapshot names.

zap plays nice with manually created snapshots or snapshots from other
tools.  It will only operate on snapshots it creates.
```

**Comment by clausecker** [comment]
@jehops Thank you for your concerns.

The current comment was supposed to read:

    COMMENT = File systems and utilities

but I must have dropped the second half during copy editing.  I'll restore that in my next push.  I am open to having a category for only file system drivers, but I would prefer to also put the fs utilities in there.  We could also add two categories: one for file system drivers and one for associated utilities (e.g. `fsutil` or `fs-mgmt`).

I'll go add the other two ports, too.

**Comment by Jehops** [comment]
> ```
> COMMENT = File systems and utilities
> ```
Taking inspiration from existing comments, how about `File systems and related utilities`? 

```
games/Makefile    COMMENT = Games and related software
databases/Makefile    COMMENT = Databases and related software
finance/Makefile    COMMENT = Monetary, financial, and related applications
```

**Comment by clausecker** [comment]
@jehops Yeah sure, works for me.  I'll put that in with my next edit.

**Comment by diizzyy** [comment]
Why this is submitted to GitHub? It's not a supported way of submitting patches (see Porters Handbook). Submit it to bugzilla and continue the discussion there.

**Comment by bsdimp** [comment]
@diizzyy Other than "the rules" is there some good reason not do it here? It seems to have generated a reasonable amount of feedback so far.

**Comment by diizzyy** [comment]
@bsdimp A lot more eyes and participants on bugzilla (and indirectly mailing lists), there's little to no involvement/interraction with most of the community including committers etc on GitHub.

**Comment by bsdimp** [comment]
@diizzyy And bugzilla is the worst tool ever to review changes (even worse than phabricator). It's barely adequate as a conveyer belt for submissions. Github is much better, and can as easily be references in other forum as bugzilla. The current bugzilla workflow is unustainable, imho, and requires too much extra work. fuz's changes likely need detailed commentary, for which bugzilla is ill suited to do on a change this large. It can't handle 10 commits in one go, let alone 140. Since it is an imperfect fit, fuz will also need to seek review elsewhere (which I thought I'd seen an email go by that did), true, but going to bugzilla is setting him up to fail because it would be such a bad tool for a change like this.

**Comment by yurivict** [comment]
> And bugzilla is the worst tool ever to review changes [...]

This is off-topic here, but you probably have never seen Jira that is used by many companies.
Compared to Jira, Bugzilla is a totally sane and pleasant tool.

--------------------------------------------

The key to your submission is not the exact list of ports, but the portmgr approval.
You need to contact somebody in the portmgr team and ask for approval.

Here is my proposal for the physics and chemistry categories from 2018 which is still not approved: https://reviews.freebsd.org/D13481

The FreeBSD Portmgr is a bit slow, don't you think?

**Comment by clausecker** [comment]
@yurivict My original email went to ports@ as well as portmgr@ and explicitly asked for approval.  I hope that suffices.  If not, I'll bug portmgr some more (I encourage you to do the same about your proposal).

**Comment by diizzyy** [comment]
@bsdimp 
Your personal preferrence is another topic as Yuri pointed out. The main issue is that this isn't a supported way of doing submissions, notifying people etc.

**Comment by bsdimp** [comment]
> @bsdimp Your personal preferrence is another topic as Yuri pointed out. The main issue is that this isn't a supported way of doing submissions, notifying people etc.

Yes. And that's a problem too. The ossified FreeBSD processes aren't helpung the project.

**Comment by clausecker** [comment]
I have just requesed an exp-run on Bugzilla: https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=281988

**Comment by mat813** [comment]
Please rename the category from `fs` to `filesystems`

**Comment by clausecker** [comment]
@mat813 That's really quite a mouthful.  Should we also rename `sysutils` to `system-utilities`?  Or `devel` to `development`?  I would like to avoid such a long category name.  FS is a well known and unambiguous acronym, so I think it's fine as a category name.

**Comment by diizzyy** [comment]
As much as fs is likely to be obvious for advanced users have really short acronyms can also cause unnecessary confusion.
OpenWrt for example also uses the full name, https://github.com/openwrt/packages/blob/master/utils/exfatprogs/Makefile#L24

**Comment by dlangille** [comment]
I too prefer filesystems over fs.

**Comment by Jehops** [comment]
I argue that using `fs` to describe the category follows precedence more than `filesystems`. 

The closest word to `file systems` in an existing category is `windows managers`, and we use `wm` for that category.   Using `fs` to describe a file system is so common it's in the first sentence of [the File Systems Wikipedia page](https://en.wikipedia.org/wiki/File_system).

**Comment by dlangille** [comment]
> I argue that using `fs` to describe the category follows precedence more than `filesystems`.
> 
> The closest word to `file systems` in an existing category is `windows managers`, and we use `wm` for that category. Using `fs` to describe a file system is so common it's in the first sentence of [the File Systems Wikipedia page](https://en.wikipedia.org/wiki/File_system).

I concede.

**Comment by clausecker** [comment]
@mat813 @Jehops @dlangille So, what do we do about this one?  Typing out filesystems makes me sad...

**Comment by mat813** [comment]
Please name the category filesystems.

**Comment by diizzyy** [comment]
> I argue that using `fs` to describe the category follows precedence more than `filesystems`.
> 
> The closest word to `file systems` in an existing category is `windows managers`, and we use `wm` for that category. Using `fs` to describe a file system is so common it's in the first sentence of [the File Systems Wikipedia page](https://en.wikipedia.org/wiki/File_system).

That being said, wm is prefixed with x11-wm for clarity

File: misc/e2fsprogs-libblkid/Makefile [hunk 1]
```diff
@@ -11,7 +11,7 @@ LIB_DEPENDS=	libuuid.so:misc/e2fsprogs-libuuid
 
 CONFIGURE_ARGS=	--enable-elf-shlibs --disable-libuuid --enable-libblkid
 
-MASTERDIR=	${.CURDIR}/../../sysutils/e2fsprogs-core
+MASTERDIR=	${.CURDIR}/../../filesystems/e2fsprogs-core
```
Comment (mat813) [inline misc/e2fsprogs-libblkid/Makefile:14]: I feel this should probably be moved to filesystems too.

File: misc/e2fsprogs-libuuid/Makefile [hunk 2]
```diff
@@ -10,7 +10,7 @@ LICENSE_FILE=	${INSTALL_WRKSRC}/COPYING
 CONFIGURE_ARGS=	--enable-elf-shlibs --enable-libuuid
 
 OPTIONS_EXCLUDE=	NLS
-MASTERDIR=	${.CURDIR}/../../sysutils/e2fsprogs-core
+MASTERDIR=	${.CURDIR}/../../filesystems/e2fsprogs-core
```
Comment (mat813) [inline misc/e2fsprogs-libuuid/Makefile:13]: I feel this should probably be moved to filesystems too.

File: security/vuxml/vuln/2011.xml [hunk 3]
```diff
@@ -1790,7 +1790,7 @@
   </vuln>
 
   <vuln vid="e55f948f-d729-11e0-abd1-0017f22d6707">
-    <topic>security/cfs -- buffer overflow</topic>
+    <topic>filesystems/cfs -- buffer overflow</topic>
```
Comment (mat813) [inline security/vuxml/vuln/2011.xml:1793]: Historical data, please revert.

File: security/vuxml/vuln/2020.xml [hunk 4]
```diff
@@ -4329,7 +4329,7 @@
   </vuln>
 
   <vuln vid="2ed7e8db-e234-11ea-9392-002590bc43be">
-    <topic>sysutils/openzfs-kmod -- critical permissions issues</topic>
+    <topic>filesystems/openzfs-kmod -- critical permissions issues</topic>
```
Comment (mat813) [inline security/vuxml/vuln/2020.xml:4332]: Historical data, please revert.

File: sysutils/automount/Makefile [hunk 5]
```diff
@@ -17,14 +17,14 @@ NO_ARCH=	yes
 OPTIONS_DEFINE=		NTFS3G EXT4 EXFAT HFS
 OPTIONS_DEFAULT=	NTFS3G EXT4 EXFAT HFS
 NTFS3G_DESC=		Enable NTFS write support with ntfs-3g over FUSE
```
Comment (mat813) [inline sysutils/automount/Makefile:19]: It should probably be moved to filesystems too, or at least get the category.

File: sysutils/dsbmd/Makefile [hunk 6]
```diff
@@ -32,12 +32,12 @@ LKL_DESC=	Btrfs and XFS support through Linux Kernel Library
 NTFS_DESC=	Mount NTFS filesystems with read and write support
 PTP_DESC=	Picture Transfer Protocol support
```
Comment (mat813) [inline sysutils/dsbmd/Makefile:33]: It should probably be moved to filesystems too, or at least get the category.

**Review by mat813 — CHANGES_REQUESTED** [review]

File: Mk/bsd.port.mk [hunk 7]
```diff
@@ -2583,7 +2583,7 @@ VALID_CATEGORIES+= accessibility afterstep arabic archivers astro audio \
 	benchmarks biology budgie cad chinese comms converters \
 	databases deskutils devel dns docs \
 	editors education elisp emulators enlightenment finance french ftp \
-	games geography german gnome gnustep graphics \
+	filesystems games geography german gnome gnustep graphics \
```
[the remark this answers is not in the document]
Comment (clausecker) [reply Mk/bsd.port.mk:2586]: Should each category be a line on its own or should only `filesystems` be a line on its own?

File: filesystems/webdavfs/Makefile [hunk 8]
```diff
@@ -1,7 +1,8 @@
-PORTNAME=	fusefs-webdavfs
+PORTNAME=	webdavfs
 DISTVERSION=	g20200520
 PORTREVISION=	23
-CATEGORIES=	sysutils
+CATEGORIES=	filesystems
+PKGNAMEPREFIX=	fusefs-
```
[the remark this answers is not in the document]
Comment (clausecker) [reply filesystems/webdavfs/Makefile:5]: This is the same change that takes place for all `fusefs-...` ports to keep us from repeating `fusefs-` all over the place in the ports tree while keeping the package name the same.  Why do you want me to revert the change for this port only?

File: filesystems/ext2/Makefile [hunk 9]
```diff
@@ -1,8 +1,9 @@
-PORTNAME=	fusefs-ext2
+PORTNAME=	ext2
 DISTVERSIONPREFIX=	v
 DISTVERSION=	0.0.11
 PORTREVISION=	2
-CATEGORIES=	sysutils
+CATEGORIES=	filesystems
+PKGNAMEPREFIX=	fusefs-
```
[the remark this answers is not in the document]
Comment (clausecker) [reply filesystems/ext2/Makefile:6]: This is the same change as for all the other fuse file systems: fusefs- is removed from the port name and turned into a prefix to the package name.  Also note that the name already doesn't agree with upstream (fuse-ext2 vs. fusefs-ext2).  Are you sure you want me to make this one port inconsistent with the others?

File: misc/e2fsprogs-libuuid/Makefile (hunk 2 above)
Comment (clausecker) [reply misc/e2fsprogs-libuuid/Makefile:13]: I had decided against moving this port in particular as it's not really about file systems, but rather just about UUID generation.

File: misc/e2fsprogs-libblkid/Makefile (hunk 1 above)
Comment (clausecker) [reply misc/e2fsprogs-libblkid/Makefile:14]: Moved to filesystems as requested.

File: sysutils/dsbmd/Makefile (hunk 6 above)
Comment (clausecker) [reply sysutils/dsbmd/Makefile:33]: Moved to filesystems as requested.

File: Mk/bsd.port.mk (hunk 7 above)
[the remark this answers is not in the document]
Comment (mat813) [reply Mk/bsd.port.mk:2586]: It's mostly one line per letter

File: filesystems/ext2/Makefile (hunk 9 above)
[the remark this answers is not in the document]
Comment (mat813) [reply filesystems/ext2/Makefile:6]: Please, keep the changes to a minimum, you are doing a sweeping commit for the category change, not fixing individual ports for other cosmetic changes.

File: filesystems/ntfs/files/patch-configure [hunk 10]
```diff
@@ -14,7 +14,7 @@
  		with_fuse="external"
  		;;
 -	freebsd*)
--		as_fn_error $? "Please see FreeBSD support at http://www.freshports.org/sysutils/fusefs-ntfs" "$LINENO" 5
+-		as_fn_error $? "Please see FreeBSD support at http://www.freshports.org/filesystems/ntfs" "$LINENO" 5
```
[the remark this answers is not in the document]
Comment (mat813) [reply filesystems/ntfs/files/patch-configure:17]: The patch-configure patch is removing lines, if you change the lines it removes, it won't match, and fail.

File: filesystems/webdavfs/Makefile (hunk 8 above)
[the remark this answers is not in the document]
Comment (mat813) [reply filesystems/webdavfs/Makefile:5]: Same answer, keep changes to a minimum

File: filesystems/ext2/Makefile (hunk 9 above)
[the remark this answers is not in the document]
Comment (clausecker) [reply filesystems/ext2/Makefile:6]: Ok, I will submit these two changes separately, even though it makes no sense.

File: Mk/bsd.port.mk (hunk 7 above)
[the remark this answers is not in the document]
Comment (clausecker) [reply Mk/bsd.port.mk]: Moved to its own line as requested.

**Comment by clausecker** [comment]
Rebased onto ab6304e314b6d17d5103606e138cf4b9d2a6e24b.

**Comment by mat813** [comment]
Can you request an exp-run so we make sure everything is squared away, and if it comes back without problems and nothing new comes up here, feel free to commit it.

**Comment by mat813** [comment]
Oh, before you land this, please squash it all into a single commit.

**Comment by clausecker** [comment]
@mat813 Are you sure?  I specifically made it a bunch of individual commits so it's easy to move individual ports back if desired.  Note that each of the commits is self-contained.

**Comment by mat813** [comment]
Please squash it all.
Also, make sure the tree is not broken with running at least `make index`.

**Comment by clausecker** [comment]
@mat813 I have verified that `make index` is fine, as is a `poudriere bulk -an` dry-run (which should pick up any dangling dependencies).

**Comment by clausecker** [comment]
Updated to just one squashed commit as requested.

Note that I'll likely have to rebase the patch set again before landing it, and that's much easier to do on the non-squashed patch set, so when landing, I will first rebase the unsquashed patch set and then squash it again before landing.

**Comment by dlangille** [comment]
> Updated to just one squashed commit as requested.
> 
> Note that I'll likely have to rebase the patch set again before landing it, and that's much easier to do on the non-squashed patch set, so when landing, I will first rebase the unsquashed patch set and then squash it again before landing.

Do you have an ETA on landing please?

**Comment by clausecker** [comment]
@dlangille I plan to land this one the instant I get portmgr approval.  There is nothing else left to do.

**Comment by rene0** [comment]
Approved with portmgr hat on

**Comment by clausecker** [comment]
Will remove the following two ports from the patch set: funionfs, fusepak.  This is because they don't have `TIMESTAMP` in their `distinfo` and they don't fetch, so I can't regenerate `distinfo`.  Both are slated for removal, so I guess it's fine.

**Comment by clausecker** [comment]
Closed by 6e2da9672f79f44048d597f0f61e4646cdeade9d.

**Comment by bdrewery** [comment]
I'm in a minority here but don't like this kind of moving chairs change.
- Some of us use "meta packages" that have RUN_DEPENDS on specific packages. Poudriere doesn't support MOVED there (no tool does). I'll consider adding support.
- Most ports reasonably could have multiple categories. What's a filesystem category that's not a sysutils? My port zfstools fits both. Are we nuking sysutils?
- "src-mgmt" is coming next, again every single tool going into there also fits in "devel", so what benefit is there?
- I don't see the motivation. What benefit does it bring? I used to be able to identity fusefs ports with `echo */*fusefs*` but now can't with that, so it's back to freshports or `git grep` anyway.
- POLA used to mean something. This kind of thing and renaming services `s/-/_/g` just creates downstream problems.

**Comment by grahamperrin** [comment]
> I'm in a minority here but don't like this kind of moving chairs change.

> … Complainers don't always show up in a PR. This change hasn't even hit quarterly yet.

I'm not averse to change, however (in a future ideal world) there should be broader planning – including `src` (at least, manual pages), not forgetting Bugzilla and FreshPorts (its [bug-finder icon](https://www.freshports.org/faq.php#bugs-find)) – before move-related commits such as this.

That's an ideal world. I'm not complaining about reality :-)

Thanks

**Comment by grahamperrin** [comment]
[283881 – Update documentation/ and website/ following moves of 142 ports to the filesystems category](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=283881)

**Pull request closed without merging**